modules/dcache-frontend/src/main/java/org/dcache/restful/resources/migration/MigrationResources.java

```diff
@@ -0,0 +1,259 @@
+            checkAppendString("state", "state", commandStrBuilder, fileAttributes);
+            checkAppendString("rp", "rp", commandStrBuilder, fileAttributes);
+            checkAppendString("size", "size", commandStrBuilder, fileAttributes);
+            if (fileAttributes.has("sticky")) {
+                // If the array exists we add the sticky flag, then if it has elements it is restricted to certain owners.
+                commandStrBuilder.append("-sticky");
+                JSONArray sticky = fileAttributes.getJSONArray("sticky");
+                if (sticky.length() > 0) {
+                    commandStrBuilder.append("=").append(sticky.getString(0));
+                }
+                // Then we can add all the rest with the delimiter
+                for (int i = 1; i < sticky.length(); i++) {
+                    commandStrBuilder.append(",").append(sticky.getString(i));
+                }
+            }
+            checkAppendString("storage", "storage", commandStrBuilder, fileAttributes);
+        }
+        // Add the target pools
+        // We know that the length != 0 from earlier!
+        for (int i = 0; i < targetPools.length(); i++) {
+            String poolName = targetPools.getString(i);
+            PoolSelectionUnit.SelectionPool sp = psu.getPool(poolName);
+            if (sp == null) {
+                throw new BadRequestException(
+                      "One of the specified pools '" + poolName + "' could not be found.");
+            }
+            commandStrBuilder.append(" ").append(poolName);
+        }
+
+        LOGGER.info("Command parsed from json payload is: '{}'.", commandStrBuilder);
+        LOGGER.info("Sending parsed command to pool '{}'.", sourcePool);
+        try {
+            String migrationResponseMsg = poolmanager.sendAndWait(
+                  new CellPath(sourcePool.getAddress()), commandStrBuilder.toString(),
+                  String.class);
+            // At the moment the command will return the migration job id with the following format:
+            // [MIG. JOB ID] STATE COMMAND
+            // Let's extract the Mig Job Id and return it.
+            LOGGER.info("Received '{}' back from source pool.", migrationResponseMsg);
+            // Get the first part of the message [MIG. JOB ID] and then remove the first and last char.
+            String digitsPart = migrationResponseMsg.split(" ")[0];
+            String digits = digitsPart.substring(1, digitsPart.length() - 1);
+            long migrationJobId = Long.parseLong(digits);
+
+            return Response.status(Response.Status.CREATED)
+                  .header("migration-job-id", migrationJobId).type(MediaType.APPLICATION_JSON)
+                  .build();
+        } catch (InterruptedException | CacheException | NoRouteToCellException e) {
+            throw new InternalServerErrorException(e);
+        }
+
+    }
+
+    private static void checkAppendString(String cmdParam, String jsonKey, StringBuilder sb,
```

Comment:
let call in `conditionalAppend` as check typically throws an exception if check failes

Reply:
Okay sounds good, will rename